lib/ace/ext/papeeria/spellchecker.coffee

```diff
@@ -37,196 +35,206 @@ define((require, exports, module) ->
-          wordRange.end.column--
-        editor.session.replace(wordRange, correction)
-        return
-
-      @init()
-      return
 
-    # Convert an array of string to popup eligible structure.
-    # @param {Array} corrections: array of strings with substitution options.
-    # @returns {Array}: array of jsons, actually.
-    convertCorrectionList = (corrections) ->
-      options = []
-      options.push({caption: item, value: item}) for item in corrections
-      return options
-
-    # Returns Range object that describes current word position.
-    # @param {Range}: current word range.
-    getCurrentWordRange = ->
-      session = editor.session
-      row = editor.getCursorPosition().row
-      col = editor.getCursorPosition().column
-      return session.getAWordRange(row, col)
-
-    # Get the word under the cursor.
-    # @returns {String}
-    extractWord = ->
-      session = editor.session
-      row = editor.getCursorPosition().row
-      wordRange = getCurrentWordRange()
-      start = wordRange.start.column
-      end = wordRange.end.column
-      # getAWordRange returns start and end positions with a trailing
-      # whitespace at the end (if there's a one), that's why replace is used
-      return session.getTextRange(wordRange).replace(/\s\s*$/, '')
+  # Sets up spellchecker popup routine and implements
+  exports.setupSpellCheckerPopup = (editor) ->
+    PopupManager.init(editor)
 
     # Check if current word is in corrections list.
     # Call showPopup if it's present, do nothing otherwise.
-    tryPopup = ->
-      word = extractWord()
+    newPopup = ->
+      word = extractWord(editor)
       spellChecker = new SpellChecker()
       correctionsItem = spellChecker.getCorrections(word)
       if correctionsItem
-        showPopup(convertCorrectionList(correctionsItem))
+        PopupManager.show(convertCorrectionList(correctionsItem))
       return
 
-    # Bind a shortcut to tryPopup callback.
+    # Bind newPopup to Alt-Enter editor shortcut
     editor.commands.addCommand({
-      name: "test",
+      name: "newPopup",
       bindKey: "Alt-Enter",
-      exec: tryPopup
+      exec: newPopup
     })
-
     return
 
+
+  # Get the word under the cursor.
+  # @param {Editor} editor: editor object
+  # @return {String}
+  extractWord = (editor) ->
+    session = editor.session
+    row = editor.getCursorPosition().row
+    wordRange = getCurrentWordRange(editor)
+    # getAWordRange returns start and end positions with a trailing
+    # whitespace at the end (if there's a one), that's why replace is used
+    return session.getTextRange(wordRange).replace(/\s\s*$/, '')
+
+  # Returns Range object that describes current word position.
+  # @param {Editor} editor: editor object
+  # @return {Range}: current word range.
+  getCurrentWordRange = (editor) ->
+    session = editor.session
+    row = editor.getCursorPosition().row
+    col = editor.getCursorPosition().column
+    return session.getAWordRange(row, col)
```

Comment:
I think you need just `session.getWordRange`. `getAWordRange` differs only in handling trailing whitespace and it seems that you don't need it